judge/views/submission.py

```diff
@@ -37,6 +38,11 @@ def submission_related(queryset):
         .prefetch_related('contest_object__authors', 'contest_object__curators')
 
 
+class SubmissionPermissionDenied(Exception):
```

Comment:
Would it make more sense semantically to inherit from `PermissionDenied`?

Reply:
Yes, it definitely would, updated.